ui/e2e/instances/details/index.spec.ts

```diff
@@ -446,3 +464,302 @@ test("after a running instance finishes, the output tab is automatically selecte
+    })
+  ).instance;
+  await page.goto(`/n/${namespace}/instances/${instanceId}`);
+
+  const logsPanel = page.getByTestId("instance-logs-container");
+
+  // I did not find any other way to select the parent of logsPanel, it does not render a testId
+  const logsPanelParent = page.locator("div").filter({ has: logsPanel }).nth(6);
+
+  const scrollbar = page.getByTestId("instance-logs-scroll-container");
+
+  await expect(logsPanel).toBeVisible();
+
+  await expect(
+    logsPanel.locator("h3"),
+    "The headline of the logs is correct"
+  ).toContainText(`Logs for /${loggingWorkflowName}`);
+
+  await expect(
+    page.getByTestId("instance-header-container").locator("div").first()
+  ).toContainText("pending");
+
+  const entriesCounter = page.getByTestId("instance-logs-entries-counter");
+
+  await expect(
+    entriesCounter.locator("span").nth(1),
+    "There is a loading spinner"
+  ).toHaveClass(/animate-ping/);
+
+  await expect(
+    entriesCounter,
+    "While starting the workflow there are 6 log entries"
+  ).toContainText("received 6 log entries");
+
+  const resizeButton = page
+    .getByTestId("instance-logs-container")
+    .getByRole("button")
+    .nth(2);
+
+  resizeButton.hover();
+  await expect(
+    page.getByText("maximize logs"),
+    "It shows the correct text when hovering over the resize button"
+  ).toBeVisible();
+
+  const minimizedHeight = (await logsPanelParent.boundingBox())?.height;
+
+  await resizeButton.click();
+
+  const maximizedHeight = (await logsPanelParent.boundingBox())?.height;
+  if (minimizedHeight === undefined || maximizedHeight === undefined) {
+    throw new Error("could not get height of logs panel");
+  }
+
+  expect(
+    maximizedHeight / minimizedHeight,
+    "The panel is significantly bigger after maximizing"
+  ).toBeGreaterThan(1.5);
+
+  page.reload();
+
+  const currentHeightAfterReload = (await logsPanelParent.boundingBox())
+    ?.height;
+  expect(
+    currentHeightAfterReload,
+    "After reloading the page, the panel is still maximized"
+  ).toEqual(maximizedHeight);
+
+  await resizeButton.hover();
+  await expect(
+    page.getByText("minimize logs"),
+    "It shows the correct text when hovering over the resize button"
+  ).toBeVisible();
+
+  await expect(
+    scrollbar.locator("pre").locator("span").nth(0),
+    "It displays an initial log entry"
+  ).toContainText("Running state logic");
+
+  await expect(
+    scrollbar.locator("pre").locator("span").nth(15),
+    "It displays the log message from the log field in the workflow yaml"
+  ).toContainText("log-message");
+
+  await expect(
+    scrollbar.locator("pre").locator("span").last(),
+    "It displays a final log entry"
+  ).toContainText("Workflow completed");
+
+  await expect(
+    entriesCounter,
+    "When the workflow finished running there are 22 log entries"
+  ).toContainText("received 22 log entries");
+});
+
+test("the logs panel can be toggled between verbose and non verbose logs", async ({
```

Comment:
Thanks for the great suggestion!

My first reaction when you mentioned the potential bug was, that I could also first toggle the button and check for the verbose text and then toggle it and check it to not contain the verbose text anymore.
But now it has become really clear to me that in this test scenario it's not really about something that should disappear, but rather about having a different text in the text field.

I also like your solution very much because of having only one assertion instead of two. 👍  

(By the way: I think "twoNumbersAndTheLogMessage" is a funny variable name, but it's also exactly fitting, so I'll stick with it 😄)